packages/dds/map/src/directory.ts

```diff
@@ -734,11 +758,13 @@ export class SharedDirectory extends SharedObject<ISharedDirectoryEvents> implem
         this.messageHandlers.set(
             "set",
             {
-                process: (op: IDirectorySetOperation, local, localOpMetadata) => {
+                process: (msg: ISequencedDocumentMessage, op: IDirectorySetOperation, local, localOpMetadata) => {
                     const subdir = this.getWorkingDirectory(op.path) as SubDirectory | undefined;
-                    if (subdir) {
+                    // If there is pending delete op for this subDirectory, then don't apply the this op as we are going
+                    // to delete this subDirectory.
+                    if (subdir && !this.isSubDirectoryDeletePending(op.path)) {
                         const context = local ? undefined : this.makeLocal(op.key, op.path, op.value);
-                        subdir.processSetMessage(op, context, local, localOpMetadata);
+                        subdir.processSetMessage(msg, op, context, local, localOpMetadata);
```

Comment:
It feels to me (on the first look) that all operations should be handled exactly the same in when pending delete is present for any parent directory (recursively). Not sure why some operations (above) have that special handling, and some (from here down) do not.

Reply:
This is just the type of op. The above 3 ops are set/delete/clear on keys while these 2 are create/delete subdirectory and we do check these things for these 2 ops in this function: needProcessSubDirectoryOperation() in detail and take action appropriately.